src/main/webapp/app/exam/participate/exercises/quiz/quiz-exam-participation.component.ts

```diff
@@ -0,0 +1,235 @@
+        // into an array of submittedAnswer objects and save it as the submittedAnswers of the submission
+        this.submission.submittedAnswers = [];
+
+        // for multiple-choice questions
+        Object.keys(this.selectedAnswerOptions).forEach((questionID) => {
+            // find the question object for the given question id
+            const question = this.quizExercise.quizQuestions.find(function (selectedQuestion) {
+                return selectedQuestion.id === Number(questionID);
+            });
+            if (!question) {
+                console.error('question not found for ID: ' + questionID);
+                return;
+            }
+            // generate the submittedAnswer object
+            const mcSubmittedAnswer = new MultipleChoiceSubmittedAnswer();
+            mcSubmittedAnswer.quizQuestion = question;
+            mcSubmittedAnswer.selectedOptions = this.selectedAnswerOptions[questionID];
+            this.submission.submittedAnswers.push(mcSubmittedAnswer);
+        }, this);
+
+        // for drag-and-drop questions
+        Object.keys(this.dragAndDropMappings).forEach((questionID) => {
+            // find the question object for the given question id
+            const question = this.quizExercise.quizQuestions.find(function (localQuestion) {
+                return localQuestion.id === Number(questionID);
+            });
+            if (!question) {
+                console.error('question not found for ID: ' + questionID);
+                return;
+            }
+            // generate the submittedAnswer object
+            const dndSubmittedAnswer = new DragAndDropSubmittedAnswer();
+            dndSubmittedAnswer.quizQuestion = question;
+            dndSubmittedAnswer.mappings = this.dragAndDropMappings[questionID];
+            this.submission.submittedAnswers.push(dndSubmittedAnswer);
+        }, this);
+        // for short-answer questions
+        Object.keys(this.shortAnswerSubmittedTexts).forEach((questionID) => {
+            // find the question object for the given question id
+            const question = this.quizExercise.quizQuestions.find(function (localQuestion) {
+                return localQuestion.id === Number(questionID);
+            });
+            if (!question) {
+                console.error('question not found for ID: ' + questionID);
+                return;
+            }
+            // generate the submittedAnswer object
+            const shortAnswerSubmittedAnswer = new ShortAnswerSubmittedAnswer();
+            shortAnswerSubmittedAnswer.quizQuestion = question;
+            shortAnswerSubmittedAnswer.submittedTexts = this.shortAnswerSubmittedTexts[questionID];
+            this.submission.submittedAnswers.push(shortAnswerSubmittedAnswer);
+        }, this);
+    }
+
+    /**
+     * Apply the data of the participation, replacing all old data
+     */
+    updateParticipationFromServer(participation: StudentParticipation) {
```

Comment:
I think we should rename this method as the participation is probably not coming from the server directly

Reply:
I renamed the method now to only _updateParticipation_